packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -262,74 +353,47 @@ class PerformanceMonitoring {
       return false
     }
 
-    if (duration > transactionDurationThreshold) {
-      if (__DEV__) {
-        this._logginService.debug(
-          `transaction(${tr.id}, ${tr.name}) was discarded! Transaction duration (${duration}) is greater than the transactionDurationThreshold configuration (${transactionDurationThreshold})`
-        )
-      }
-      return false
-    }
-
-    if (tr.options.managed && tr.spans.length === 0) {
-      if (__DEV__) {
-        this._logginService.debug(
-          `transaction(${tr.id}, ${tr.name}) was discarded! Transaction does not have any spans`
-        )
+    if (isManagedTransaction(tr)) {
+      if (duration > TRANSACTION_DURATION_THRESHOLD) {
+        if (__DEV__) {
+          this._logginService.debug(
+            `transaction(${tr.id}, ${tr.name}) was discarded! Transaction duration (${duration}) is greater than managed transaction threshold (${TRANSACTION_DURATION_THRESHOLD})`
+          )
+        }
+        return false
       }
-      return false
-    }
-
-    /**
-     * In case of unsampled transaction, send only the transaction to apm server
-     *  without any spans to reduce the payload size
-     */
-    if (!tr.sampled) {
-      tr.resetSpans()
-    }
-
-    if (tr.options.checkBrowserResponsiveness) {
-      const wasBrowserResponsive = this.checkBrowserResponsiveness(
-        tr,
-        BROWSER_RESPONSIVENESS_INTERVAL,
-        BROWSER_RESPONSIVENESS_BUFFER
-      )
 
-      if (!wasBrowserResponsive) {
+      if (tr.spans.length === 0) {
         if (__DEV__) {
           this._logginService.debug(
-            `transaction(${tr.id}, ${tr.name}) was discarded! Browser was not responsive enough during the transaction.`,
-            ' duration:',
-            duration,
-            ' browserResponsivenessCounter:',
-            tr.browserResponsivenessCounter
+            `transaction(${tr.id}, ${tr.name}) was discarded! Transaction does not have any spans`
           )
         }
         return false
       }
-    }
-    return true
-  }
 
-  prepareTransaction(transaction) {
-    transaction.spans.sort(function(spanA, spanB) {
-      return spanA._start - spanB._start
-    })
+      if (tr.options.checkBrowserResponsiveness) {
```

Comment:
We can remove this option here, since we're only applying it to managed transactions and there's no config option anymore.

Reply:
But,we don't want it to run for `page-load` right? So i kept it.

Reply:
Lets think of a way, we can either check type and not run the check or check if `browserResponsivenessCounter` is > 0.